python/nano/src/bigdl/nano/deps/ipex/ipex_inference_model.py

```diff
@@ -0,0 +1,132 @@
+        if self.use_ipex:
+            self.model = ipex.optimize(self.model)
+        if self.use_jit:
+            self.model = torch.jit.trace(self.model, input_sample)
+            self.model = torch.jit.freeze(self.model)
+
+    def forward_step(self, *inputs):
+        if self.channels_last:
+            inputs = tuple(map(lambda x: x.to(memory_format=torch.channels_last), inputs))
+        return self.model(*inputs)
+
+    def _save_model(self, path):
+        if self.use_jit:
+            self.model.save(path / "ckpt.pth")
+        else:
+            torch.save(self.original_state_dict, path / "ckpt.pth")
+
+
+class PytorchIPEXJITModel(IPEXJITModel, AcceleratedLightningModule):
+    def __init__(self, model, input_sample=None, use_ipex=False,
+                 use_jit=False, channels_last=None, from_load=False):
+        '''
+        This is the accelerated model for pytorch and ipex/jit.
+        All the external API is based on Trainer, so what we have here is
+        basically internal APIs and subject to change.
+
+        This PytorchIPEXJITModel will serve for fp32 and ipex>1.9 models.
+        :param model: the model(nn.module) to be transform if from_load is False
+               the accelerated model if from_load is True.
+        :param input_sample: torch tensor indicate the data sample to be used
+               for tracing.
+        :param use_ipex: if use ipex to optimize the model
+        :param use_jit: if use jit to accelerate the model
+        :param channels_last: if set model and data to be channels-last mode.
+               the parameter will be ignored if use_ipex is False.
+        :param from_load: this will only be set by _load method.
+        '''
+        AcceleratedLightningModule.__init__(self, None)
+        IPEXJITModel.__init__(self, model, input_sample=input_sample,
+                              use_ipex=use_ipex, use_jit=use_jit, from_load=from_load)
+
+    def on_forward_start(self, inputs):
+        return inputs
+
+    def on_forward_end(self, outputs):
+        return outputs
+
+    @property
+    def status(self):
+        status = super().status
+        status.update({"use_ipex": self.use_ipex,
+                       "use_jit": self.use_jit,
+                       "channels_last": self.channels_last,
+                       "checkpoint": "ckpt.pth"})
+        return status
+
+    @staticmethod
+    def _load(path, model):
+        status = PytorchIPEXJITModel._load_status(path)
+        checkpoint_path = path / status['checkpoint']
+        if status["use_jit"]:
+            model = torch.jit.load(checkpoint_path)
+            model.eval()
+            model = torch.jit.freeze(model)
+        else:
+            state_dict = torch.load(checkpoint_path)
+            model.eval()
+            model.load_state_dict(state_dict)
+            if status["channels_last"]:
+                model = model.to(memory_format=torch.channels_last)
+            model = ipex.optimize(model)
```

Comment:
I think you can just load the checkpoint and pass the model to `PytorchIPEXJITModel`.

Reply:
good catch, fixed